### PR TITLE
Degen chain support

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

### CODE REVIEW

File: src/components/expanded-state/swap-details/CurrencyTile.js [hunk 1]
```diff
@@ -96,7 +95,7 @@ export default function CurrencyTile({
                       </NativePriceText>
                     </Column>
                     <Column alignHorizontal="center" width="content">
-                      <NativePriceText>{symbol}</NativePriceText>
+                      <NativePriceText>{asset?.symbol}</NativePriceText>
```
Comment (walmat) [inline src/components/expanded-state/swap-details/CurrencyTile.js:98]: unrelated to the work on this PR, but maybe we want some fallback if for whatever reason symbol is `undefined`?

File: src/screens/ExplainSheet.js [hunk 2]
```diff
@@ -461,6 +465,19 @@ export const explainers = (params, theme) => {
       text: AVALANCHE_EXPLAINER,
       title: lang.t('explain.avalanche.title'),
     },
+    degen: {
+      emoji: '⛽️',
+      extraHeight: 144,
+      logo: <ChainBadge network={networkTypes.degen} marginBottom={8} position="relative" size="large" />,
+      readMoreLink: buildRainbowLearnUrl({
+        url: 'https://learn.rainbow.me/a-beginners-guide-to-layer-2-networks',
```
Comment (dereknelson) [inline src/screens/ExplainSheet.js:473]: looks like we're gonna need to make a new learn doc

File: src/hooks/useSwapAdjustedAmounts.ts [hunk 3]
```diff
@@ -40,7 +47,10 @@ export default function useSwapAdjustedAmounts(tradeDetails: Quote) {
     (tradeDetails.sellTokenAddress === ETH_ADDRESS && tradeDetails.buyTokenAddress === WETH_ARBITRUM_ADDRESS) ||
     // zora eth <-> weth swap
     (tradeDetails.buyTokenAddress === ETH_ADDRESS && tradeDetails.sellTokenAddress === WETH_ZORA_ADDRESS) ||
-    (tradeDetails.sellTokenAddress === ETH_ADDRESS && tradeDetails.buyTokenAddress === WETH_ZORA_ADDRESS)
+    (tradeDetails.sellTokenAddress === ETH_ADDRESS && tradeDetails.buyTokenAddress === WETH_ZORA_ADDRESS) ||
+    // degen <-> wdegen swap
+    (tradeDetails.buyTokenAddress === ETH_ADDRESS && tradeDetails.sellTokenAddress === WDEGEN_DEGEN_CHAIN_ADDRESS) ||
+    (tradeDetails.sellTokenAddress === ETH_ADDRESS && tradeDetails.buyTokenAddress === WDEGEN_DEGEN_CHAIN_ADDRESS)
```
Comment (walmat) [inline src/hooks/useSwapAdjustedAmounts.ts:53]: I remember having to touch this when we were having some swaps issues when wrapping eth to weth on zora. We need to explore a way to no longer have to add entries here.

File: src/references/index.ts [hunk 4]
```diff
@@ -88,6 +88,7 @@ export const BNB_BSC_ADDRESS = AddressZero;
 export const BNB_MAINNET_ADDRESS = '0xb8c77482e45f1f44de1745f52c74426c631bdd52';
 export const MATIC_MAINNET_ADDRESS = '0x7d1afa7b718fb893db30a3abc0cfc608aacfebb0';
 export const MATIC_POLYGON_ADDRESS = '0x0000000000000000000000000000000000001010';
+export const DEGEN_CHAIN_DEGEN_ADDRESS = AddressZero;
```
Comment (walmat) [inline src/references/index.ts:91]: do we need to add an entry here? Can't we just use AddressZero directly?

**Review by walmat — APPROVED** [review]
Looks good! Just a few smol comments.

**Review by dereknelson — APPROVED** [review]
somebody get this man to a haberdashery

File: src/components/asset-list/RecyclerAssetList2/FastComponents/FastFallbackCoinIconImage.tsx [hunk 5]
```diff
@@ -60,15 +60,19 @@ export const FastFallbackCoinIconImage = React.memo(function FastFallbackCoinIco
   );
 
   return (
-    <View style={[sx.coinIconContainer, sx.withShadow, { shadowColor, height: size, width: size }]}>
+    <View style={[sx.coinIconContainer, sx.withShadow, { shadowColor, height: size, width: size, borderRadius: size / 2 }]}>
```
Comment (skylarbarrera) [inline src/components/asset-list/RecyclerAssetList2/FastComponents/FastFallbackCoinIconImage.tsx:63]: nice

File: src/networks/degen.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+    nativeCurrency: {
+      ...degen.nativeCurrency,
+      address: DEGEN_CHAIN_DEGEN_ADDRESS,
+    },
+
+    rpc: proxyRpcEndpoint(degen.id),
+    getProvider: getProviderForNetwork(Network.degen),
+    // need to find balance checker address
+    balanceCheckerAddress: '',
```
Comment (skylarbarrera) [inline src/networks/degen.ts:33]: not needed, we could remove this prop pretty easy but not a bid deal atm

File: src/screens/ExplainSheet.js (hunk 2 above)
Comment (skylarbarrera) [reply src/screens/ExplainSheet.js:473]: yeah, can ping support to atleast add a ref for degen

**Review by skylarbarrera — APPROVED** [review]
looks great will do some testing

**Pull request merged**